tests/integrations/raftstore/test_unsafe_recovery.rs

```diff
@@ -300,6 +300,54 @@ fn test_force_leader_for_learner() {
     cluster.must_transfer_leader(region.get_id(), find_peer(&region, 1).unwrap().clone());
 }
 
+// Test the case that three of five nodes fail and force leader on a hibernated
+// previous leader.
+#[test]
+fn test_force_leader_on_hibernated_leader() {
+    let mut cluster = new_node_cluster(0, 5);
+    configure_for_hibernate(&mut cluster);
+    cluster.pd_client.disable_default_operator();
+
+    cluster.run();
+    cluster.must_put(b"k1", b"v1");
+
+    let region = cluster.get_region(b"k1");
+    cluster.must_split(&region, b"k9");
+    let region = cluster.get_region(b"k2");
+    let peer_on_store1 = find_peer(&region, 1).unwrap();
+    cluster.must_transfer_leader(region.get_id(), peer_on_store1.clone());
+
+    // wait a while to hibernate
+    std::thread::sleep(Duration::from_millis(
+        cluster.cfg.raft_store.raft_election_timeout_ticks as u64
+            * cluster.cfg.raft_store.raft_base_tick_interval.as_millis()
+            * 3,
+    ));
```

Comment:
It would be better if we assert the target region is hibernated here. Something like `assert_eq!(is_hibernated(region.get_id()), ture);`

Reply:
can't, it's not exported